src/main/scala/wdl4s/values/WdlFile.scala

```diff
@@ -35,14 +39,20 @@ sealed trait WdlFile extends WdlPrimitive {
     case _ => invalid(s"$value == $rhs")
   }
 
-  override def valueString = value.toString
+  override def valueString: String = value.toString
 }
 
-case class WdlSingleFile(value: String) extends WdlFile {
-  override def toWdlString = "\"" + value.toString + "\""
+case class WdlSingleFile(origValue: String) extends WdlFile {
+  override def toWdlString: String = "\"" + value.toString + "\""
+
+  // Validate file path
```

Comment:
@geoffjentry Yeah we realized that making it an actual URI, although good on paper turns out to be hard because of the constraints that it puts on the String that we don't necessarily want to enforce (spaces need to be encoded, no underscore in host names etc...)